### ISSUE TITLE
Accuracy

### ISSUE THREAD

**Issue by brandonros** [BrainJS/brain.js#71]
```
var net = new brain.NeuralNetwork();

var trainingSet = [
  {
    input: [0.1, 0.1],
    output: [0.2]
  },
  {
    input: [0.2, 0.2],
    output: [0.4]
  },
  {
    input: [0.5, 0.1],
    output: [0.6]
  },
  {
    input: [0, 0.1],
    output: [0.1]
  },
  {
    input: [0.3, 0.1],
    output: [0.4]
  }
];

console.log(net.train(trainingSet, { // Object {error: 0.000099992701244754, iterations: 15197}
  errorThresh: 0.0001,
  learningRate: 0.3
}));

console.log(net.run([0.2, 0.4])); // [0.5480908025594973]
console.log(net.run([0.1, 0.4])); // [0.45124782633077687]
console.log(net.run([0.3, 0.4])); // [0.6239448854007088]
console.log(net.run([0.5, 0.4])); // [0.721436435561435]
console.log(net.run([0, 0.4])); // [0.3393225809755128]
```

I've tried a few different approaches and all sorts of input/outputs (and another library actually).

I really can't get accurate data out. Any tips?

**Comment by Dok11** [comment]
@brandonros your training data looks like random numbers. Maybe they dont have any dependecies between input and outup?
For example, second input key does not affect to result in the set. And one input to one output is... strange case.

**Comment by brandonros** [comment]
It's addition.

Regards,

Brandon
________________________________

**Comment by Dok11** [comment]
> error: 0.000099992701244754

Absolutely, your net remember just training data. In other words - overtrain.
Normal error will be between 0.3 — 0.02

**Comment by robertleeplummerjr** [comment]
What was your error rate at end of training?

**Comment by brandonros** [comment]
> Absolutely, your net remember just training data. In other words - overtrain.
> Normal error will be between 0.3 — 0.02

Can you share an example on how I could achieve what I am trying to achieve (neural network learns that it's simple addition) using the input I'm using? I've tried tweaking the error rate + not overtraining, but didn't get any better results.

**Comment by wagenaartje** [comment]
@brandonros, you cannot expect a neural network to map a function with such a small dataset - you have included just 5 training samples. Note that a neural network will always minimize the error on the given dataset, so you cannot guarantee that a neural network will perform as good on a a dataset that it was not trained on.

There are infinite possibilities for addition. So _at least_, scale your dataset up to 100 samples - you can do this dynamically by generating them.  Split these 100 samples up in óne group of 80, and one group of 20.

Train the network on the 80 samples until it has reachted a low error. Then, test your neural network on the 20 samples on which it wasn't trained; if the error is low on those 20 samples, you know your neural network has successfully learned how to addition in some way or another.

**Comment by Dok11** [comment]
@brandonros as one of variants you can decrease purpose accuracy to some real data. Imagine what can somebody answer who have same dataset?
Really, I think what your data doesn't have any relationship. Or you need dramatically increase volume of trainig data as said @wagenaartje 
So, set `iterations` less then 50, or get more training dataset.

**Comment by brandonros** [comment]
@wagenaartje Thank you for taking the time to respond. One question:

Would it ever make sense to train the model with the same data in reptition? Example:

```
var trainingSet = [];

for (var i = 0; i < 20; i++) {
  trainingSet.push({
    input: [0, 0],
    output: [0]
  });
}

for (var i = 0; i < 20; i++) {
  trainingSet.push({
    input: [0.1, 0.1],
    output: [0.2]
  });
}

for (var i = 0; i < 20; i++) {
  trainingSet.push({
    input: [0.2, 0.2],
    output: [0.4]
  });
}

console.log(feedForwardNet.train(trainingSet));
```

**Comment by brandonros** [comment]
@Dok11

> Really, I think what your data doesn't have any relationship.

Sorry that I did not make it clear. The data should be representing addition. 0.1 + 0.1 = 0.2, etc.

**Comment by Dok11** [comment]
Strange, I did think what NN work just with multiply, but sum.. I don't think it easy task for NN

**Comment by robertleeplummerjr** [comment]
The datset should be fine if you give it the loose principles of math, it just needs a good deal of them.  Something I've tried that has worked nice is giving training results in every case from 0 to 10, for example.  It depends what your ceiling is and your floor, but if it is a finite set you can give something general enough but specific to not get overfitting.
